test/ca_updatedb.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifdef OPENSSL_SYS_UNIX
+# include <unistd.h>
+#endif
+#include "../apps/include/ca.h"
+#include "../apps/include/ca_logic.h"
+#include "../apps/include/apps.h"
+
+/* tweaks needed for Windows */
+#ifdef _WIN32
+# define access _access
+# define timezone _timezone
+#endif
+
+char *default_config_file = NULL;
+
+int main(int argc, char *argv[])
+{
+    CA_DB *db = NULL;
+    BIO *channel;
+    time_t *testdateutc = NULL;
+    int r;
+
+    if (argc != 3) {
+        fprintf(stderr, "Usage: %s indexfile testdate\n", argv[0]);
+        fprintf(stderr, "       testdate format: ASN1-String\n");
+	exit(EXIT_FAILURE);
```

Comment:
A direct return is enough, no need to call exit from a test fixture.

Reply:
I'm not sure what you mean. Change to "return 1"? Only concerning this exit or all exits? Would the exit cause some unwanted behavior?

Reply:
Calling `exit` from a `main` method is the same as putting a `return EXT_FAILURE' in this `main`.
Actually, the code base don't use `exit` method, a lot .